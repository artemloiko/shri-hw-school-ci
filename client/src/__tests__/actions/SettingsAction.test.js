import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {
  fetchSettingsIfNeeded,
  updateSettings,
  updateSettingsFail,
  resetSettingsError,
} from '../../actions/SettingsAction';
import apiMock from '../../utils/api';

jest.mock('../../utils/api.js');

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = {
  isFetching: false,
  isLoaded: false,
};
let store = null;

describe('settings actions', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });
  afterEach(() => {
    store = null;
  });

  test('fetching settings success', async () => {
    const data = {
      id: 'c42db8b8-128a-4194-a19b-7974cabccf4f',
      repoName: 'artuom130/shri-hw-async',
      buildCommand: 'npm run build',
      mainBranch: 'master',
      period: 0,
    };
    apiMock.getSettings.mockImplementationOnce(() => Promise.resolve(data));

    await store.dispatch(fetchSettingsIfNeeded());

    expect(store.getActions()).toMatchSnapshot();
  });
  test('fetching settings fail', async () => {
    const error = { message: 'Internal server error' };
    apiMock.getSettings.mockImplementationOnce(() => Promise.reject(error));

    await store.dispatch(fetchSettingsIfNeeded());

    expect(store.getActions()).toMatchSnapshot();
  });
  test('update settings', () => {
    const data = {
      repoName: 'artuom130/itItReal',
      buildCommand: 'npm run build',
      mainBranch: 'master',
      period: 0,
    };

    store.dispatch(updateSettings(data));

    expect(store.getActions()).toMatchSnapshot();
  });
  test('update settings error', () => {
    const error = {
      message: 'Cannot find artuom130/shri-hw-asyncs repository',
      errorCode: 'GIT_CANNOT_FIND_REPO',
    };

    store.dispatch(updateSettingsFail(error));

    expect(store.getActions()).toMatchSnapshot();
  });
  test('reset settings error', () => {
    store.dispatch(resetSettingsError());

    expect(store.getActions()).toMatchSnapshot();
  });
});
