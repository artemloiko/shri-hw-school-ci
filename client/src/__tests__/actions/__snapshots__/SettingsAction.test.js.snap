// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings actions fetching settings fail 1`] = `
Array [
  Object {
    "type": "GET_SETTINGS_REQUEST",
  },
  Object {
    "error": true,
    "payload": "Internal server error",
    "type": "GET_SETTINGS_FAIL",
  },
]
`;

exports[`settings actions fetching settings success 1`] = `
Array [
  Object {
    "type": "GET_SETTINGS_REQUEST",
  },
  Object {
    "payload": Object {
      "buildCommand": "npm run build",
      "id": "c42db8b8-128a-4194-a19b-7974cabccf4f",
      "mainBranch": "master",
      "period": 0,
      "repoName": "artuom130/shri-hw-async",
    },
    "type": "GET_SETTINGS_SUCCESS",
  },
]
`;

exports[`settings actions reset settings error 1`] = `
Array [
  Object {
    "type": "RESET_SETTINGS_ERROR",
  },
]
`;

exports[`settings actions update settings 1`] = `
Array [
  Object {
    "payload": Object {
      "buildCommand": "npm run build",
      "mainBranch": "master",
      "period": 0,
      "repoName": "artuom130/itItReal",
    },
    "type": "SET_SETTINGS",
  },
]
`;

exports[`settings actions update settings error 1`] = `
Array [
  Object {
    "payload": Object {
      "errorCode": "GIT_CANNOT_FIND_REPO",
      "message": "Cannot find artuom130/shri-hw-asyncs repository",
    },
    "type": "SET_SETTINGS_FAIL",
  },
]
`;
